### ISSUE TITLE
[Bug] duplicate inspection for "Parameter/variable is not used"

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1161]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                   |
| Plugin         | Php Inspections (EA Extended), last version |
| Language level | e.g. PHP 7                        |

Current behaviour (description/screenshot:)
--
Consider the following code:
```php
<?php

function foo ($url) {
    if (false) {

    }
    else {
        $errormsg = "Failed";
        if ( false ) {
            $errormsg .= "Failed 1";
        }
        else {
            $errormsg .=  "Failed 2";
        }
    }
}
```
It seems that the inspection is launch multiple times for the same error:
![duplicate_inspection](https://user-images.githubusercontent.com/9605520/50915086-521f0700-1438-11e9-98ac-d73c0ec57d89.PNG)

In the original code, I had the inspection launched up to 6 times for both variables.

In the inspection panel, it's actually not so bad:
![panel](https://user-images.githubusercontent.com/9605520/50915182-85619600-1438-11e9-8a4c-0885a95afb3a.PNG)

But in the XML output files PHPStorm export, we find the duplicates again.

Expected behaviour
--
The inspection should only be fired once for every variable and the XML output file should only contain a single occurence of it.

Additionnal details
I'm trying to build a tool to parse the xml files and report resolved inspections and new inspections between each import of the files. This bug is very problematic as the inspections are not even sorted the same way as two consecutive inspections would be and it mess up my tool.

Thanks!

**Comment by kalessil** [comment]
Declining, this is desired behaviour - it helps to see which code branches are affected.

**Comment by kalessil** [comment]
Oh, addressing this should not be a problem!

**Comment by orklah** [comment]
Great! Thanks!

**Comment by orklah** [comment]
Hey, sorry to bother you but this issue is quite critical for my project. Do you know when you will be able to look at it?

Thanks!

**Comment by kalessil** [comment]
Moved on the top of issues list. I'll try to fix it Mon/Tue, will poke you with details where to take a fresh plugin version.

**Comment by orklah** [comment]
Great, thank you!

**Comment by kalessil** [comment]
Hey @orklah, which PhpStorm version is it (I can not reproduce the issue for some reason)?

**Comment by kalessil** [comment]
@orklah: can you try this binary in your environment and share feedback if the issue has been fixed ([binary](https://github.com/kalessil/phpinspectionsea/raw/bug-1161/PhpInspectionsEA.jar), btw do you know how to install plugin from the local file?)?

**Comment by orklah** [comment]
Hi, Thanks, i'll try that right away!

Yes, I already installed from local file a few times.

**Comment by orklah** [comment]
I still have the issue.

I can reproduce it very well on the attached file (I had to rename to .txt for github)
[rss.txt](https://github.com/kalessil/phpinspectionsea/files/2876659/rss.txt)
and it generate the xml attached (had to rename as well)
[OnlyWritesOnParameterInspection.txt](https://github.com/kalessil/phpinspectionsea/files/2876674/OnlyWritesOnParameterInspection.txt)

It is just at the beggining of the file:
```xml
<problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>512</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
  <problem>
  <file>file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php</file>
  <line>509</line>
  <module>inspections_parser</module>
  <entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/analyze_zone/wordpress-5.0.2-fr_FR/wordpress/wp-includes/rss.php" />
  <problem_class severity="WEAK WARNING" attribute_key="NOT_USED_ELEMENT_ATTRIBUTES">Parameter/variable is not used</problem_class>
  <description>Parameter/variable is overridden, but is never used or appears outside of the scope.</description>
</problem>
```

There is a variable on line 509 and another on line 512 and each are reported 6 times. But in the file it's 512-512-509-509-512-512...


Thanks!

**Comment by kalessil** [comment]
Ok, able to reproduce now, checking.

**Comment by kalessil** [comment]
Fixed now, I updated the binary (please ensure the plugin version will be 3.0.10.2)

**Comment by orklah** [comment]
There are less cases but still some on the same lines
[OnlyWritesOnParameterInspection.txt](https://github.com/kalessil/phpinspectionsea/files/2881515/OnlyWritesOnParameterInspection.txt)

Is there something you can do?

**Comment by kalessil** [comment]
@orklah : is the last comment still relevant? If so, please, file a new bug.
